### DISCUSSION TITLE
如何自定义 Markup/AP 扩展简化 Behavior.InvokeCommandAction？

### DISCUSSION THREAD

**Discussion by CodingOctocat** [BYJRK/DotNet-Discussions#80] [WPF 开发]
使用 Behaviors 有个麻烦的地方，只是单纯的绑定应该事件命令，却需要 5 行代码，能否自定义 Markup/AP 扩展简化代码？

常规用法(代码仅供演示，Button “碰巧” 有 `Command` 对应 `Click` 事件)：
```XAML
<Button x:Name="button">
  <Behaviors:Interaction.Triggers>
    <Behaviors:EventTrigger EventName="Click" SourceObject="{Binding ElementName=button}">
      <Behaviors:InvokeCommandAction Command="{Binding ColorCommand}" CommandParameter="{Binding Grid.Background}" />
    </Behaviors:EventTrigger>
 </Behaviors:Interaction.Triggers>
</Button>
```

期望用法：
```XAML
<Button x:Name="button" ix:Button.Click="{Binding 💡ColorCommand, CommandParameter="{Binding Grid.Background}}"/>
```

不过自己写的 Markup 似乎不会有智能感知，也就是说期望用法代码的灯泡💡位置不会有完成列表选择 `ColorCommand`，需要自己完整的敲出 `ColorCommand` 单词。

**Comment by BYJRK** [comment]
这种花括号的都是 Markup 语法，你可以试试自己写一个类，继承 MarkupExtension，然后为 Button 添加一个附加属性，并在 OnPropertyChangedCallback 中实现为 Button 添加 Behavior 的逻辑

**Reply by CodingOctocat** [reply]
重点在于智能感知，我问了一圈，`XamlBehaviors` 其中的一名贡献者可能会去改进这一点，不过智能感知是个麻烦的活。
